src/service/action-impl.js

```diff
@@ -287,15 +287,21 @@ export class ActionService {
         }
       });
       this.root_.addEventListener('keydown', event => {
-        const element = dev().assertElement(event.target);
-        const {key} = event;
+        const {key, target} = event;
+        const element = dev().assertElement(target);
         if (key == Keys.ENTER || key == Keys.SPACE) {
           const role = element.getAttribute('role');
           const isTapEventRole =
               (role && hasOwn(TAPPABLE_ARIA_ROLES, role.toLowerCase()));
           if (!event.defaultPrevented && isTapEventRole) {
-            event.preventDefault();
-            this.trigger(element, name, event, ActionTrust.HIGH);
+            const actionInvoked =
+                this.trigger(element, name, event, ActionTrust.HIGH);
+            // Only if an action was invoked do we prevent the default.
+            // In the absence of an action, e.g. on="[event].method", we do not
+            // want to stop default behavior.
+            if (actionInvoked) {
```

Comment:
Per prior comment, this can be named "hasAction".

Reply:
right. Will make the change once I get your input on my suggestion in your first comment. Thanks